crates/transaction-pool/src/maintain/drift_monitor.rs

```diff
@@ -0,0 +1,321 @@
+    pub fn add_dirty_addresses<I>(&mut self, addresses: I)
+    where
+        I: IntoIterator<Item = Address>,
+    {
+        self.dirty_addresses.extend(addresses);
+    }
+
+    /// Set the set of dirty addresses
+    pub fn set_dirty_addresses(&mut self, addresses: HashSet<Address>) {
+        self.dirty_addresses = addresses;
+    }
+
+    /// Take the set of dirty addresses, leaving the internal set empty
+    pub fn take_dirty_addresses(&mut self) -> HashSet<Address> {
+        std::mem::take(&mut self.dirty_addresses)
+    }
+
+    /// Remove an address from the set of dirty addresses
+    pub fn remove_dirty_address(&mut self, address: &Address) -> bool {
+        self.dirty_addresses.remove(address)
+    }
+
+    /// Returns `true` if the monitor has dirty addresses to reload
+    pub fn has_dirty_addresses(&self) -> bool {
+        !self.dirty_addresses.is_empty()
+    }
+
+    /// Start reloading accounts from state
+    pub fn start_reload_accounts<Client, Spawner>(
+        &mut self,
+        client: Client,
+        at: BlockHash,
+        task_spawner: &Spawner,
+    ) where
+        Client: StateProviderFactory + Clone + 'static,
+        Spawner: reth_tasks::TaskSpawner + 'static,
+    {
+        if !self.has_dirty_addresses() || !self.reload_accounts_fut.is_terminated() {
+            return;
+        }
+
+        let (tx, rx) = oneshot::channel();
+
+        let fut = if self.dirty_addresses.len() > self.max_reload_accounts {
+            // need to chunk accounts to reload
+            let accs_to_reload = self
+                .dirty_addresses
+                .iter()
+                .copied()
+                .take(self.max_reload_accounts)
+                .collect::<Vec<_>>();
+
+            for acc in &accs_to_reload {
+                // make sure we remove them from the dirty set
+                self.dirty_addresses.remove(acc);
+            }
+
+            async move {
+                let res = load_accounts(client, at, accs_to_reload);
+                let _ = tx.send(res);
+            }
+            .boxed()
+        } else {
+            // can fetch all dirty accounts at once
+            let accs_to_reload = std::mem::take(&mut self.dirty_addresses);
+
+            async move {
+                let res = load_accounts(client, at, accs_to_reload);
+                let _ = tx.send(res);
+            }
+            .boxed()
+        };
+
+        self.reload_accounts_fut = rx.fuse();
+        task_spawner.spawn_blocking(fut);
+    }
+
+    /// update metrics based on the current state
+    pub fn update_metrics(&self) {
+        self.metrics.set_dirty_accounts_len(self.dirty_addresses.len());
+    }
+}
+
+impl Future for DriftMonitor {
+    type Output = DriftMonitorResult;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        // Check if reload future is ready
+        if !self.reload_accounts_fut.is_terminated() {
```

Comment:
I think we can skip this because fuse should always return pending if terminated?